contracts/0.8.9/lib/PositiveTokenRebaseLimiter.sol

```diff
@@ -0,0 +1,136 @@
+      * @param _rebaseLimit max limiter value (saturation point), see `LIMITER_PRECISION_POINTS`
+      * @param _totalPooledEther total pooled ether, see `Lido.getTotalPooledEther()`
+      * @param _totalShares total shares, see `Lido.getTotalShares()`
+      * @return limiterState newly initialized limiter structure
+      */
+    function initLimiterState(
+        uint256 _rebaseLimit,
+        uint256 _totalPooledEther,
+        uint256 _totalShares
+    ) internal pure returns (TokenRebaseLimiterData memory limiterState) {
+        if(_rebaseLimit == 0) revert TooLowTokenRebaseLimit();
+        if(_rebaseLimit > UNLIMITED_REBASE) revert TooHighTokenRebaseLimit();
+
+        // special case
+        if(_totalPooledEther == 0) { _rebaseLimit = UNLIMITED_REBASE; }
+
+        limiterState.totalPooledEther = _totalPooledEther;
+        limiterState.totalShares = _totalShares;
+        limiterState.rebaseLimit = _rebaseLimit;
+    }
+
+    /**
+     * @notice check if positive rebase limit is reached
+     * @param _limiterState limit repr struct
+     * @return true if limit is reached
+     */
+    function isLimitReached(TokenRebaseLimiterData memory _limiterState) internal pure returns (bool) {
+        return _limiterState.accumulatedRebase == _limiterState.rebaseLimit;
+    }
+
+    /**
+     * @notice raise limit using the given amount of ether
+     * @param _limiterState limit repr struct
+     */
+    function raiseLimit(TokenRebaseLimiterData memory _limiterState, uint256 _etherAmount) internal pure {
+        if(_limiterState.rebaseLimit == UNLIMITED_REBASE) { return; }
+
+        uint256 projectedLimit = _limiterState.rebaseLimit + (
+            _etherAmount * LIMITER_PRECISION_BASE
+        ) / _limiterState.totalPooledEther;
+
+        _limiterState.rebaseLimit = Math256.min(projectedLimit, UNLIMITED_REBASE);
+    }
+
+    /**
+     * @dev append ether and return the consumed value not exceeding the limit
+     * @param _limiterState limit repr struct
+     * @param _etherAmount desired ether addition
+     * @return consumedEther allowed to add ether to not exceed the limit
+     */
+    function consumeLimit(TokenRebaseLimiterData memory _limiterState, uint256 _etherAmount)
+        internal
+        pure
+        returns (uint256 consumedEther)
+    {
+        if (_limiterState.rebaseLimit == UNLIMITED_REBASE) {
+            return _etherAmount;
+        }
+
+        uint256 remainingRebase = _limiterState.rebaseLimit - _limiterState.accumulatedRebase;
+        uint256 remainingEther = (remainingRebase * _limiterState.totalPooledEther) / LIMITER_PRECISION_BASE;
+
+        consumedEther = Math256.min(remainingEther, _etherAmount);
+
+        if (consumedEther == remainingEther) {
+            _limiterState.accumulatedRebase = _limiterState.rebaseLimit;
+        } else {
+            _limiterState.accumulatedRebase += (
+                consumedEther * LIMITER_PRECISION_BASE
+            ) / _limiterState.totalPooledEther;
```

Comment:
I guess it can't be, see:
```solidity
        // special case
        if(_totalPooledEther == 0) { _rebaseLimit = UNLIMITED_REBASE; }

        limiterState.totalPooledEther = _totalPooledEther;
        limiterState.totalShares = _totalShares;
        limiterState.rebaseLimit = _rebaseLimit;
```

and 

```solidity
    function consumeLimit(TokenRebaseLimiterData memory _limiterState, uint256 _etherAmount)
        internal
        pure
        returns (uint256 consumedEther)
    {
        if (_limiterState.rebaseLimit == UNLIMITED_REBASE) {
            return _etherAmount;
        }
 ```

Reply:
Agree, missed it